src/test/java/com/google/firebase/database/core/RepoTest.java

```diff
@@ -346,6 +387,82 @@ public void onComplete(DatabaseError error, boolean committed, DataSnapshot curr
     assertEquals(update, events.get(0).getValue());
   }
 
+  @Test
+  public void testTransactionError() throws InterruptedException {
+    final Repo repo = newRepo();
+    final List<DataSnapshot> events = new ArrayList<>();
+    final Path path = new Path("/txn_error");
```

Comment:
I'm gonna use my knowledge that I just learned from Testing on the Toilet:

I think these tests would be easier to follow if you set up the mocks inline.

Reply:
Done